app/controllers/application_controller.rb

```diff
@@ -22,6 +22,11 @@ class ApplicationController < Sinatra::Base
   # This will be a ajax call
   post '/' do
     @bin = Bin.new(params[:bin])
+    # calculate the expiration date - now + retention_time in minutes
+    if params.dig(:retention)&.to_i&.> 0
+      @bin.expire_date = Time.now + params[:retention].to_i.minutes
+      params.delete(:retention)
```

Comment:
Should we set a maximum limit for retentions? Otherwise it is possible to create datasets that live almost forever

Reply:
good point.

The database inserts the default (7 days), if no `expire_date` is given,
but without a max-check this is not enough.